tests/hooks/test_http_hook.py

```diff
@@ -334,5 +338,35 @@ def test_connection_without_host(self, mock_get_connection):
         hook.get_conn({})
         self.assertEqual(hook.base_url, 'http://')
 
+    @requests_mock.mock()
+    def test_post_json_request(self, mock_requests):
+        obj1 = {'a': 1, 'b': 'abc', 'c': [1, 2, {"d": 10}]}
+        obj2 = [1, 2, 3]
+
+        # Ensure that obj1 was encoded to JSON
+        def match_obj1(request):
+            return json.loads(request.text) == obj1
+
+        # Filters to catch posted JSON
+        mock_requests.post(
+            '//test:8080/v1/test',
+            status_code=200,
+            request_headers={'Content-Type': 'application/json'},
+            text='test_post_json_request',
+            additional_matcher=match_obj1
+        )
+
+        with mock.patch(
+            'airflow.hooks.base_hook.BaseHook.get_connection',
+            side_effect=get_airflow_connection
+        ):
+            # Send obj1 as JSON and verify it matched the mock
+            resp = self.post_hook.run('v1/test', json=obj1)
+            self.assertEqual(resp.status_code, 200)
+            self.assertEqual(resp.text, 'test_post_json_request')
+        # Ensure that obj1 was what was sent
+        with self.assertRaises(requests_mock.exceptions.NoMockAddress):
+            resp = self.post_hook.run('v1/test', json=obj2)
```

Comment:
seems you are just verifying here that the request mock raises with bad address...  does this belong in `test_post_json_request`?

Reply:
I wanted to be sure that obj1 was encoded & posted correctly, not just that a post was done. 
So I needed to check the contents of the request payload - using additional_matcher to ensure that what was posted matched. However since request_mock is a bit confusing to me I wasnt convinced that additional_matcher was going to be used. So I added a check to post obj2 the same way as obj1, which raises an exception

Reply:
ok i was confused by the fact that you are looking for `NoMockAddress` error it's a little misleading but i now understand that's how requests mock is designed.

but anyway, so you are trying to test the test here.  i think you should chop this.  just test your feature, don't test the functionality of requests mock.  fine to do it locally to satisfy yourself that it's working right, but i don't think it belongs in here permanently.

that said... this test-of-the-test is working right anyway, because on line 368 if you replace `obj2` with `obj1`, the test still passes.  this is because you aren't mocking `airflow.hooks.base_hook.BaseHook.get_connection` at this point of the code, so it is finding no match, irrespective of the object you supply.  


2. i think you should chop this testing of the test.  just satisfy yourself that the test is doing

Reply:
Interesting, thanks for testing that. Requests_mock is tricky to get my head around. Anyway the second test is a bit overkill and if it isn't doing what I thought then it's certainly worth removing it :)